Core.Auditing/IfxAuditLogger.cs

```diff
@@ -61,36 +60,46 @@ public void LogApplicationAuditEvent(ITelemetryClient telemetryClient, AuditMand
+                OperationName = tokenType + TokenGenerationOperation,
+            };
+            auditMandatoryProperties.AddAuditCategory(AuditEventCategory.Authorization);
+            LogAuditEvent(telemetryClient,  targetResources, callerIdentities, auditMandatoryProperties, auditOptionalProperties);
         }
 
         public void LogCertificateFetchAuditEvent(ITelemetryClient telemetryClient, OperationResult operationResult, TargetResource[] targetResources, CallerIdentity[] callerIdentities, AuditOptionalProperties auditOptionalProperties = null)
         {
-            LogAuthorizationAuditEvent(telemetryClient, operationResult, FetchCertificateOperation, targetResources, callerIdentities, auditOptionalProperties);
+            AuditMandatoryProperties auditMandatoryProperties = new AuditMandatoryProperties()
+            {
+                ResultType = operationResult,
+                OperationName = FetchCertificateOperation
+            };
+            auditMandatoryProperties.AddAuditCategory(AuditEventCategory.Authorization);
+            LogAuditEvent(telemetryClient, targetResources, callerIdentities, auditMandatoryProperties, auditOptionalProperties);
+        }
+
+        public void LogRequest(ITelemetryClient telemetryClient, TargetResource[] targetResources, CallerIdentity[] callerIdentities, AuditMandatoryProperties auditMandatoryProperties, AuditOptionalProperties auditOptionalProperties = null)
+        {
+            auditMandatoryProperties.AddAuditCategory(AuditEventCategory.Other);
+            LogAuditEvent(telemetryClient, targetResources, callerIdentities, auditMandatoryProperties, auditOptionalProperties);
         }
 
-        private void LogAuthorizationAuditEvent(ITelemetryClient telemetryClient, OperationResult operationResult, string operationName, TargetResource[] targetResources, CallerIdentity[] callerIdentities, AuditOptionalProperties auditOptionalProperties)
+        private void LogAuditEvent(ITelemetryClient telemetryClient, TargetResource[] targetResources, CallerIdentity[] callerIdentities, AuditMandatoryProperties auditMandatoryProperties, AuditOptionalProperties auditOptionalProperties)
         {
             string webAppName = Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME");
             if (string.IsNullOrEmpty(webAppName))
             {
-                telemetryClient?.LogWarning($"[{nameof(LogAuthorizationAuditEvent)}] Web app name isn't found from environment variable.");
+                telemetryClient?.LogWarning($"[{nameof(LogAuditEvent)}] Web app name isn't found from environment variable.");
                 webAppName = DefaultWebAppName; // Set to default web app name.
             }
-
-            var auditMandatoryProperties = new AuditMandatoryProperties
-            {
-                OperationName = operationName
-            };
-
-            auditMandatoryProperties.AddAuditCategory(AuditEventCategory.Authorization);
```

Comment:
Instead of pushing this to the callers, I would pass the audit category as an argument to the method so that it is done here (instead of 3 times above).